libr/bin/p/bin_bf.c

```diff
@@ -64,32 +61,43 @@ static RBinInfo *info(RBinFile *bf) {
 	return ret;
 }
 
-static bool check_bytes(const ut8 *buf, ut64 length) {
-	int i, is_bf = 0;
-	if (buf && length > 0) {
-		int max = R_MIN (16, length);
-		const char *p = (const char *) buf;
-		is_bf = 1;
-		for (i = 0; i < max; i++) {
-			switch (p[i]) {
-			case '+':
-			case '-':
-			case '>':
-			case '<':
-			case '[':
-			case ']':
-			case ',':
-			case '.':
-			case ' ':
-			case '\n':
-			case '\r':
-				break;
-			default:
-				is_bf = 0;
-			}
+static bool check_buffer(RBuffer *buf) {
+	r_return_val_if_fail (buf, false);
+
+	ut8 tmp[16];
+	int read_length = r_buf_read_at (buf, 0, tmp, sizeof (tmp));
+	if (read_length <= 0) {
```

Comment:
what about != 16

Reply:
well != sizeof(tmp).

Reply:
I'm trying to keep the original semantic. And originally the check was `length > 0`, it wasn't really a requirement that 16 bytes were read.